src/components/DateRangeInput/OverlayComponent.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import styled from '@emotion/styled';
+import Box from '../Box';
+import Card from '../Card';
+import Button from '../Button';
+import Flex from '../Flex';
+import TimePicker from './TimePicker';
+import { IconButton } from '../../index';
+
+interface CustomOverlayProps {
+  month?: Date;
+  date?: { from: Date; to: Date };
+  selectedDay?: Date;
+  withTime?: boolean;
+  goToNextMonth: () => void;
+  onCancel: () => void;
+  onApply: (date?: Date) => void;
+  onTimeUpdate: (date: { from?: Date; to?: Date }) => void;
+}
+
+const Wrapper = styled(Box)`
+  position: relative;
+
+  .DayPicker-Day {
+    position: relative;
+    z-index: 3;
+    .day {
+      position: relative;
+      z-index: 4;
+    }
+    .bg,
```

Comment:
We need to add some empty divs as we also need to create a continuous background along with the highlighted start and ending date.